src/EditorFeatures/Core/Implementation/Diagnostics/DiagnosticsSquiggleTaggerProvider.TagSource.cs

```diff
@@ -61,25 +62,48 @@ protected override TagSpan<IErrorTag> CreateTagSpan(SnapshotSpan span, Diagnosti
                 return new TagSpan<IErrorTag>(span, new ErrorTag(errorType, diagnostic.Message));
             }
 
-            private string GetErrorTypeFromDiagnosticKind(DiagnosticData diagnostic)
+            private string GetErrorTypeFromDiagnostic(DiagnosticData diagnostic)
             {
-                if (diagnostic.CustomTags.Count > 1)
+                return GetErrorTypeFromDiagnosticTags(diagnostic) ??
+                       GetErrorTypeFromDiagnosticProperty(diagnostic) ??
+                       GetErrorTypeFromDiagnosticSeverity(diagnostic);
+            }
+
+            private string GetErrorTypeFromDiagnosticProperty(DiagnosticData diagnostic)
+            {
+                if (diagnostic.Properties.Count == 0)
+                {
+                    return null;
+                }
+
+                string value;
+                if (!diagnostic.Properties.TryGetValue(WellKnownDiagnosticPropertyNames.Origin, out value))
+                {
+                    return null;
+                }
+
+                if (value == WellKnownDiagnosticTags.Build && _blueSquiggleForBuildDiagnostic)
+                {
+                    return PredefinedErrorTypeNames.CompilerError;
+                }
+
+                return null;
+            }
+
+            private string GetErrorTypeFromDiagnosticTags(DiagnosticData diagnostic)
+            {
+                if (diagnostic.CustomTags.Count <= 1)
```

Comment:
Why <=? If we just have one tag and that's the one we are interested in, we will return null here.